### PR TITLE
feat(ci): Introduce `zepter` checks

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Introduces `zepter` checks, with a configuration derived from [`alloy`](https://github.com/alloy-rs/alloy). [`zepter`](https://crates.io/crates/zepter) enforces correct feature propagation and formatting.

### CODE REVIEW

File: .config/zepter.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        "propagate-feature",
+        # These are the features to check:
+        "--features=arbitrary,std,serde,test-utils,metrics",
+        # Do not try to add a new section into `[features]` of `A` only because `B` exposes that feature. There are edge-cases where this is still needed, but we can add them manually.
+        "--left-side-feature-missing=ignore",
+        # Ignore the case that `A` is outside of the workspace. Otherwise it will report errors in external dependencies that we have no influence on.
+        "--left-side-outside-workspace=ignore",
+        # Only check normal dependencies.
+        # Propagating to dev-dependencies leads to compilation issues: https://github.com/alloy-rs/alloy/pull/2144#issuecomment-2702291349
+        "--dep-kinds=normal:check,dev:ignore",
```
Comment (refcell) [inline .config/zepter.yaml:22]: Can we build this set of checks ourselves instead of cp from alloy

Comment (clabby) [reply .config/zepter.yaml:22]: Yeah for sure. Drafted for now as I experiment

File: .github/workflows/rust_ci.yaml [hunk 2]
```diff
@@ -228,3 +228,20 @@ jobs:
...
+  zepter:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+      - name: Install zepter
+        run: |
+          cargo install zepter -f --locked
+          zepter --version
```
Comment (refcell) [inline .github/workflows/rust_ci.yaml:241]: good candidate for a github action

File: crates/supervisor/rpc/Cargo.toml [hunk 3]
```diff
@@ -34,8 +34,14 @@ op-alloy-consensus.workspace = true
...
+	"alloy-eips/serde",
+	"alloy-primitives/serde",
+	"kona-interop/serde",
+	"kona-protocol/serde",
+	"op-alloy-consensus/serde",
+]
 # The `jsonrpsee` feature enables the core RPC functionality.
 # When it's active, we also need the `serde` feature.
-jsonrpsee = ["serde", "dep:jsonrpsee"]
-client = ["jsonrpsee", "jsonrpsee/client"] # Client feature builds upon the base jsonrpsee feature
```
Comment (refcell) [inline crates/supervisor/rpc/Cargo.toml:41]: Ah it's failing to parse out comments. Maybe we can shift this to a line above?

**Review by refcell — APPROVED** [review]
